subprojects/core/src/main/java/org/gradle/internal/progress/DefaultBuildOperationExecutor.java

```diff
@@ -124,6 +128,8 @@ public void run(RunnableBuildOperation buildOperation) {
     }
 
     private <O extends BuildOperation> void executeInParallel(BuildOperationQueue.QueueWorker<O> worker, Action<BuildOperationQueue<O>> queueAction) {
+        failIfInResourceLockTransform();
```

Comment:
Good point on the performance implications.  I've added the exception to both scenarios.